OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -88,6 +89,12 @@ public override IEnumerable<Target> GetTargets(Actor self)
 		{
 			yield return target;
 		}
+
+		public override TargetLineNode TargetLineNode(Actor self)
+		{
+			var color = NextActivity == null ? Color.Green : NextActivity.TargetLineNode(self).Color;
```

Comment:
Looking at the last node isn't correct. If I issue an attack order then make the unit afterwards, everything will be drawn in green. That's why I didn't get your struct approach. Two solutions, I think.

(1) Giveup color altogether during waypoint mode and paint everything blue. (2) Revert to color property in older commit